engine/src/main/java/org/terasology/world/internal/EntityAwareWorldProvider.java

```diff
@@ -124,6 +124,21 @@ public Block setBlock(Vector3i pos, Block type) {
         return null;
     }
 
+    @Override
+    public Map<Vector3i, Block> setBlocks(Map<Vector3i, Block> blocks) {
+        if (GameThread.isCurrentThread()) {
```

Comment:
Might it be worth adding a quick comment to a line or two here explaining the issue, or even overriding / appending some Javadoc in this case?